### PR TITLE
Fix style updates to avoid unnecessary allocation

### PR DESCRIPTION
**Pull request by dead-claudia**
## Description

Fix style updates to avoid an unnecessary intermediate object. I recast it locally and had the code ready prior to #2311, which should explain why it's a little less surgical.

- Drive-by: properly censor `cssFloat` -> css `float:` to match DOM behavior

## Motivation and Context

See [this comment](https://github.com/MithrilJS/mithril.js/pull/2311#issuecomment-441641636). This is a follow-up to #2311.

## How Has This Been Tested?

Ran the usual and nothing broke. ☺️

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation change

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] I have updated `docs/change-log.md`

### CODE REVIEW

File: render/render.js [hunk 1]
```diff
@@ -775,38 +775,43 @@ module.exports = function($window) {
...
 	function updateStyle(element, old, style) {
-		if (old != null && style != null && typeof old === "object" && typeof style === "object" && style !== old) {
+		if (old === style) {
+			// Do nothing.
+		} else if (style == null) {
+			element.style.cssText = ""
+		} else if (typeof style !== "object") {
+			element.style.cssText = style
+		} else if (old == null || typeof old !== "object") {
+			element.style.cssText = ""
```
Comment (StephanHoyer) [inline render/render.js]: Position of this line confused me first. Maybe move below the comment.

File: render/render.js [hunk 2]
```diff
@@ -775,38 +775,43 @@ module.exports = function($window) {
...
-			? prop
-			: prop.replace(matchUpperCase, prependDashAndLowerCase)
-	}
-
 	//style
+	var uppercaseRegex = /[A-Z]/g
+	function toLowerCase(capital) { return "-" + capital.toLowerCase() }
+	function normalizeKey(key) {
+		return key[0] === "-" && key[1] === "-" ? key :
+			key === "cssFloat" ? "float" :
```
Comment (StephanHoyer) [inline render/render.js]: is this tested?

Comment (isiahmeadows) [reply render/render.js]: Oops. Fixed.

File: render/render.js (hunk 1 above)
Comment (isiahmeadows) [reply render/render.js]: Fixed. I also need to add a few more comments around that area.

Comment (isiahmeadows) [reply render/render.js]: *needed

**Comment by isiahmeadows** [comment]
@StephanHoyer These updates look good to you?

**Pull request merged**